### PR TITLE
Get glslify working with ify-loader

### PR DESCRIPTION
**Pull request by rreusser**
It seems that `opts._flags` is empty by default. When this gets passed to `glslify@5` that's fine. `glslify@6` fails because that causes it to fall into the string output branch. The fix is to simply specify `opts._flags = opts._flags || []`. That causes it to function as a transform.

Things to do:
- [x] test it
- [x] ensure `[]` is the proper default for `opts._flags`. (This seems not particularly important, maybe? It's *a* default that works fine since it's only relevant when it was unspecified and just needed to be truthy. It could be `foo` and work just fine, right? But should still check.)
- [x] The glslify output fixture has an extra comma at the end. Is that normal?

I'm not sure it's strictly "correct", but since glslify makes the presence of `opts._flags` a hard requirement in order for it to function as a transform, it seems reasonable that it's okay to just provide that as a courtesy for all transforms.

cc (and big thanks to) @bpostlethwaite @verazab

See also: #5, #7, #9, https://github.com/glslify/glslify/issues/74

### CODE REVIEW

**Comment by rreusser** [comment]
Confirmed working. You'll have to hook it up to this particular PR (or just add that flag manually in `node_modules/ify-loader/index.html`), but with this change, this repo compiles both just fine and produces plotly plots: https://github.com/rreusser/plotly-webpack

**Comment by VeraZab** [comment]
I can test webpack 1 & 2

**Comment by VeraZab** [comment]
tested with webpack: 2.7.0 & 1.15.0: works!

**Comment by bpostlethwaite** [comment]
Thanks @VeraZab and @rreusser. Is this ready to merge?

**Comment by bpostlethwaite** [comment]
@hughsk did you want to take a look? I'll hold off merging for 3 days to give you some time. We'll test it out on a few more transforms in the meantime.

File: fixtures/glsl/output.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+#define GLSLIFY 1
+#define GLSLIFY 1
+vec4 ones () {
+  return vec4(1);
+}
+
+void main () {
+  gl_FragColor = ones();
+}
+,
```
Comment (rreusser) [inline fixtures/glsl/output.txt]: I don't know why this comma exists in the output, but perhaps it's nothing problematic.

**Comment by rreusser** [comment]
From the [browserify docs on opts](https://github.com/browserify/browserify/blob/0305b703b226878f3acb5b8f2ff9451c87cd3991/readme.markdown#btransformtr-opts):

> Options sent to the browserify constructor are also provided under opts._flags. These browserify options are sometimes required if your transform needs to do something different when browserify is run in debug mode, for example.

So my feeling is that there are two approaches to that:

1. Require the user to provide dummy opts that make the transform behave correctly. I don't like this option because unless it's documented very prominently, it's pretty hard to catch, and also puts the burden on the user to obey a random quirk.
2. This PR fills in dummy opts. The only downside here is that it could *potentially* cause a breaking change with some unknown transform, though it seems unlikely. More likely is that the breaking change is for it to now work, as for glslify. This might justify a minor version bump.

**Comment by rreusser** [comment]
@bpostlethwaite Yes, I cleaned up a couple small things. I'm content.

**Pull request merged**